packages/core/package-manager/src/NodePackageManager.js

```diff
@@ -71,15 +71,55 @@ export class NodePackageManager implements PackageManager {
     |},
   ): Promise<any> {
     let {resolved} = await this.resolve(name, from, opts);
-    return this.load(resolved, from);
+    return await this.load(resolved, from);
   }
 
   requireSync(name: ModuleSpecifier, from: FilePath): any {
     let {resolved} = this.resolveSync(name, from);
-    return this.load(resolved, from);
+    return this.loadSync(resolved, from);
   }
 
-  load(resolved: FilePath, from: FilePath): any {
+  async load(resolved: FilePath, from: FilePath): any {
+    if (!path.isAbsolute(resolved)) {
+      // Node builtin module
+      // $FlowFixMe
+      return require(resolved);
+    }
+
+    let filePath = await this.fs.realpath(resolved);
+    const cachedModule = Module._cache[filePath];
+    if (cachedModule !== undefined) {
+      return cachedModule.exports;
+    }
+
+    let m = new Module(filePath, Module._cache[from] || module.parent);
+    Module._cache[filePath] = m;
+
+    // Patch require within this module so it goes through our require
+    m.require = async id => {
```

Comment:
Does this actually work? Doesn't `require()` return a promise now instead of the exports?

Reply:
Reverted this part. There must be a workaround. 

https://github.com/parcel-bundler/parcel/pull/5642/commits/ce0f80291bfc7704d85d12b732fc03a2fcd5893b

Reply:
https://github.com/jaydenlin/node-async-require